test/performance/range/gap_decorator_rand_read_benchmark.cpp

```diff
@@ -0,0 +1,93 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <chrono>
+#include <cmath>
+#include <cstring>
+#include <random>
+#include <utility>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alignment/exception.hpp>
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/range/decorator/all.hpp>
+#include <seqan3/std/ranges>
+
+#include "gap_decorator_helper.hpp"
+
+using namespace seqan3;
+
+// Apply benchmarks with custom ranges for grid parameters sequence length and gap proportions
+static void CustomArguments(benchmark::internal::Benchmark* b) {
+    std::array<long long int, 5> gap_percentages = {1, 5, 25, 50, 75};
+    for (long long int seq_len = 4; seq_len <= (1 << 18); seq_len <<= 2)
+    {
+        for (auto gap_percentage : gap_percentages)
+            b->Args({seq_len, gap_percentage});
+    }
+}
```

Comment:
please make this conform to our style-guide ( snake_case, no static, { to newline, fixed width integers...)